proposals/NNNN-New-vehicle-data-StabilityControlsStatus.md

```diff
@@ -0,0 +1,172 @@
+* Author: [Ankur Tiwari](https://github.com/atiwari9)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core | HMI | Policy Server | SHAID | iOS | Java Suite]
+
+## Introduction
+
+This proposal is to add new vehicle data `StabilityControlsStatus` to know if stability controls like ESC, Traction control etc. are active.
+
+## Motivation
+
+In order to partner with more diverse app partners, we need to provide additional sets of vehicle data items through SDL. This goes in line with commitment to enhance SDL with even richer vehicle data content.
+
+## Proposed Solution 
+
+We need to add `StabilityControlsStatus` for `GetVehicleData`, `SubscribeVehicleData`, `UnsubscribeVehicleData` & `OnVehicleData` RPCs. Following are the changes needed in API:
+
+### Updates in MOBILE_API:
+
+#### Add to enum `VehicleDataType`: 
+
+```xml
+<element name="VEHICLEDATA_STABILITYCONTROLSSTATUS" since="X.x"/>
+```
+#### Add new struct `StabilityControlsStatus`:
+
+```xml	
+<struct name="StabilityControlsStatus" since="x.x">
+	<param name="escSystem" type="VehicleDataStatus" mandatory="false">
+		<description>true if vehicle stability control is ON, else false</description>
+	</param>
+	<param name="trailerSwayControl" type="VehicleDataStatus" mandatory="false">
+		<description>true if vehicle trailer sway control is ON, else false</description>
+	</param>
+</struct>
+
+```
+#### Add to function `SubscribeVehicleData` request: 
+
+```xml	
+<param name="stabilityControlsStatus" type="Boolean" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `SubscribeVehicleData` response: 
+
+```xml	
+<param name="stabilityControlsStatus" type="VehicleDataResult" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `UnsubscribeVehicleData` request: 
+
+```xml	
+<param name="stabilityControlsStatus" type="Boolean" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `UnsubscribeVehicleData` response: 
+
+```xml	
+<param name="stabilityControlsStatus" type="VehicleDataResult" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `GetVehicleData` request: 
+
+```xml	
+<param name="stabilityControlsStatus" type="Boolean" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `GetVehicleData` response: 
+
+```xml	
+<param name="stabilityControlsStatus" type="StabilityControlsStatus" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
+#### Add to function `OnVehicleData` response: 
+
+```xml	
+<param name="stabilityControlsStatus" type="StabilityControlsStatus" mandatory="false" since="X.x">
+	<description>See StabilityControlsStatus</description>
+</param>
+```
```

Comment:
Can this section be re-organized to be a bit cleaner and easier to read?  Perhaps group all requests together and all responses together?

For example:

### Updates in MOBILE_API:

Add the following parameter to these function requests:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml	
<param name="stabilityControlsStatus" type="Boolean" mandatory="false" since="X.x">
	<description>See StabilityControlsStatus</description>
</param>
```

Add the following parameter to these function responses:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml	
<param name="stabilityControlsStatus" type="VehicleDataResult" mandatory="false" since="X.x">
	<description>See StabilityControlsStatus</description>
</param>
```

Reply:
Done